### ISSUE TITLE
WarGames Crashes

### ISSUE THREAD

**Issue by Destova** [elishacloud/dxwrapper#61]
I have absolutely zero idea if this is where I should be asking this, but I finally decided to give your WarGames Windows 10 patch a try, and at first all seemed well. However, after trying to put a hacker into a computer center in the second human level, it crashed for the first time, and now it will randomly crash just about any time as I try to get back to the second level. All I can really see is it suddenly spikes my GPU to 100% and then desktops me.

**Comment by elishacloud** [comment]
Can you upload your log file?  I will take a look at this when I get a chance.

You could also try disabling `Dd7to9` and enabling `DDrawCompatExperimental` to see if that helps.

**Comment by Destova** [comment]
I just tested those two changes and it seems they work. I'm not able to do an indepth test as I'm currently working from home and can't exactly play video games just now but I'll post later if I can make it past the second mission.

I wanted to ask though, I don't see a log file anywhere. Where would I find that?

**Comment by elishacloud** [comment]
The log file should be called `dxwrapper-wargames.log` and it should be in the game folder.

**Comment by Destova** [comment]
Well as far as I can tell your instructions fixed the issue. Was able to get to mission 3 before I had to stop for the night, farther than I've gotten in two decades. Music doesn't work for very long though, but that seems to be the single only issue, so, thank you very very much for the patch and for the fix!

**Comment by mirh** [comment]
So.. how is it going by now? Everything as it should in the end?

**Comment by Destova** [comment]
> So.. how is it going by now? Everything as it should in the end?

Yes! I played through several missions and it worked perfectly. If any further issues arise I'll let you know but as of right now, everything works well.

**Comment by LWalker85** [comment]
I've gone through the fix and am still not able to even launch the game. I'm not sure if I've put the patch files in the wrong place,  but I open the file as an admin, compatibility off, run through the rasterizer and get a black screen, then back to my desktop. Any thoughts as where to start?

**Comment by elishacloud** [comment]
Have you tried with the latest update [here](https://github.com/elishacloud/dxwrapper/wiki/WarGames)?  It should be working now without crashing.

**Comment by elishacloud** [comment]
This issue should be fixed with the latest release:  https://github.com/elishacloud/dxwrapper/wiki/WarGames

**Comment by Destova** [comment]
Sorry to necro a dead thread, but I'm now having a new issue? 

I actually got this working about 18 days ago, ran just fine, even streamed it via twitch with zero issues. 

But today, as I'm trying to help a friend potentially start this thing up, it now refuses to boot for me. 

I am running on Windows 11, not Windows 10, which your patch says should work either way, but I'm now stumped. I'm getting a pop up after selecting software rasterizer, that says "An error occurred while starting the game. Please check your installation and setup."
![image](https://github.com/user-attachments/assets/b4222b1d-d127-42d2-81f4-ede6b24dac4e)

My log is as below:

> 35060 23:54:42.650 Starting DxWrapper v1.1.6900.22
35060 23:54:42.650 Running from: C:\WarGames\dxwrapper.dll
35060 23:54:42.650 Reading config file: C:\WarGames\dxwrapper.ini
35060 23:54:42.650 ASRock B650E PG Riptide WiFi Default string (Desktop)
35060 23:54:42.650 ASRock B650E PG Riptide WiFi Default string (Desktop)
35060 23:54:42.650 NVIDIA GeForce RTX 3080
35060 23:54:42.650 Windows 10 Enterprise 64-bit (10.0.22621)
35060 23:54:42.650 wargames.exe (PID:28100)
35060 23:54:42.655 Environment variable __COMPAT_LAYER = ""
35060 23:54:42.655 Utils::WriteMemory::WriteMemory Writing bytes to memory...
35060 23:54:42.655 Disabling High DPI Scaling...
35060 23:54:42.655 Loaded library: user32.dll
35060 23:54:42.655 Loaded library: shcore.dll
35060 23:54:42.655 Setting SingleCoreAffinity...
35060 23:54:42.655 Utils::GetProcessMask Setting CPU mask: 0x2
35060 23:54:42.655 Set exception handler
35060 23:54:42.655 Loaded library: kernel32.dll
35060 23:54:42.655 Finished setting exception handler
35060 23:54:42.655 Loading 'ddraw.dll'...
35060 23:54:42.655 Hooking ddraw.dll APIs...
35060 23:54:42.656 Loaded library: d3d9.dll
35060 23:54:42.656 Enabling ddraw wrapper
35060 23:54:42.656 Enabling d3d9 wrapper
35060 23:54:42.656 Loading 'd3d9.dll'...
35060 23:54:42.656 Loaded library: C:\WINDOWS\system32\d3d9.dll
35060 23:54:42.656 DxWrapper loaded!
35060 23:54:42.670 dd_DirectDrawEnumerateA
35060 23:54:42.670 d9_Direct3DCreate9
35060 23:54:42.670 Redirecting 'Direct3DCreate9' ...
35060 23:54:42.710 Creating interface m_IDirect3D9Ex::m_IDirect3D9Ex (036AB258)
35060 23:54:42.710 m_IDirect3D9Ex::LogAdapterNames Adapter: 0 \\.\DISPLAY1 NVIDIA GeForce RTX 3080
35060 23:54:42.710 m_IDirect3D9Ex::LogAdapterNames Adapter: 1 \\.\DISPLAY2 NVIDIA GeForce RTX 3080
35060 23:54:42.710 dd_DirectDrawCreate
35060 23:54:42.710 dd_DirectDrawCreateEx
35060 23:54:42.710 Redirecting 'DirectDrawCreate' IID_IDirectDraw to --> 'Direct3DCreate9'
35060 23:54:42.710 Creating interface m_IDirectDrawX::m_IDirectDrawX (036AD958) converting interface from v1 to v9
35060 23:54:42.710 Redirecting 'Direct3DCreate9' ...
35060 23:54:42.713 Creating interface m_IDirect3D9Ex::m_IDirect3D9Ex (036AB408)
35060 23:54:42.750 m_IDirectDrawX::~m_IDirectDrawX (036AD958) deleting interface!
35060 23:54:42.750 m_IDirect3D9Ex::~m_IDirect3D9Ex (036AB408) deleting interface!
35060 23:54:42.750 m_IDirectDrawSurfaceX::CleanupSharedEmulatedMemory Deleting 0 emulated surfaces!
35060 23:54:42.750 m_IDirect3D9Ex::~m_IDirect3D9Ex (036AB258) deleting interface!
35060 23:54:44.223 Redirecting 'DirectDrawCreate' IID_IDirectDraw to --> 'Direct3DCreate9'
35060 23:54:44.223 Creating interface m_IDirectDrawX::m_IDirectDrawX (009DA740) converting interface from v1 to v9
35060 23:54:44.223 Redirecting 'Direct3DCreate9' ...
35060 23:54:44.256 Creating interface m_IDirect3D9Ex::m_IDirect3D9Ex (037132E8)
35060 23:54:44.279 m_IDirectDrawX::~m_IDirectDrawX (009DA740) deleting interface!
35060 23:54:44.280 m_IDirect3D9Ex::~m_IDirect3D9Ex (037132E8) deleting interface!
35060 23:54:44.280 m_IDirectDrawSurfaceX::CleanupSharedEmulatedMemory Deleting 0 emulated surfaces!
35060 23:55:44.719 Quiting DxWrapper
35060 23:55:44.719 Unloading libraries...
35060 23:55:44.719 Unloading exception handlers
35060 23:55:44.719 DxWrapper terminated!


Hoping you can help, cause like I said this was running successfully not even a few weeks ago on the same PC, same everything, but now I'm at a loss.

**Comment by elishacloud** [comment]
I don't have Windows 11 to test with.  But you can try the latest build to see if this works:  [dxwrapper.zip](https://github.com/user-attachments/files/16277442/dxwrapper.zip)

**Comment by Destova** [comment]
> I don't have Windows 11 to test with. But you can try the latest build to see if this works: [dxwrapper.zip](https://github.com/user-attachments/files/16277442/dxwrapper.zip)

I am using the latest build of it, it's doing what I described above.

**Comment by elishacloud** [comment]
From the log files the build you were using is `35060 23:54:42.650 Starting DxWrapper v1.1.6900.22`.  The latest build is `v1.1.7109.22`.  Did you try the new build I gave you?

**Comment by Destova** [comment]
Oh snap I didn't realize you sent a direct file. My bad, give me like, half an hour and I'll report back on the results. My apologies.

**Comment by Destova** [comment]
> From the log files the build you were using is `35060 23:54:42.650 Starting DxWrapper v1.1.6900.22`. The latest build is `v1.1.7109.22`. Did you try the new build I gave you?

All right, so tried it with that version you sent. It almost works, but goes to a black screen for a moment, then back to desktop. New log below:

> 6368 17:17:06.564 Starting DxWrapper v1.1.7072.22
6368 17:17:06.565 Running from: C:\WarGames\dxwrapper.dll
6368 17:17:06.565 Reading config file: C:\WarGames\dxwrapper.ini
6368 17:17:06.565 ASRock B650E PG Riptide WiFi Default string (Desktop)
6368 17:17:06.565 ASRock B650E PG Riptide WiFi Default string (Desktop)
6368 17:17:06.565 NVIDIA GeForce RTX 3080
6368 17:17:06.565 Windows 10 Enterprise 64-bit (10.0.22621)
6368 17:17:06.565 wargames.exe (PID:18012)
6368 17:17:06.569 Environment variable __COMPAT_LAYER = ""
6368 17:17:06.569 WriteMemory::WriteMemory Writing bytes to memory...
6368 17:17:06.569 Disabling High DPI Scaling...
6368 17:17:06.569 Loaded library: user32.dll
6368 17:17:06.569 Loaded library: shcore.dll
6368 17:17:06.569 Setting SingleCoreAffinity...
6368 17:17:06.569 Utils::GetProcessMask Setting CPU mask: 0x2
6368 17:17:06.569 Set exception handler
6368 17:17:06.569 Loaded library: kernel32.dll
6368 17:17:06.569 Finished setting exception handler
6368 17:17:06.569 Loading 'ddraw.dll'...
6368 17:17:06.569 Hooking ddraw.dll APIs...
6368 17:17:06.570 Installing GDI & User32 hooks
6368 17:17:06.570 Loaded library: gdi32.dll
6368 17:17:06.570 Loaded library: d3d9.dll
6368 17:17:06.570 Enabling ddraw wrapper
6368 17:17:06.570 Enabling d3d9 wrapper
6368 17:17:06.570 Loading 'd3d9.dll'...
6368 17:17:06.570 Loaded library: C:\WINDOWS\system32\d3d9.dll
6368 17:17:06.570 Loaded library: winmm.dll
6368 17:17:06.570 Installing memory management hooks
6368 17:17:06.601 Installing messaging hooks
6368 17:17:06.601 Installing display mode hooks
6368 17:17:06.601 Installing registry hooks
6368 17:17:06.601 Installing Win32 hooks
6368 17:17:06.603 DxWrapper loaded!
6368 17:17:06.617 dd_DirectDrawEnumerateA
6368 17:17:06.617 d9_Direct3DCreate9
6368 17:17:06.617 Redirecting 'Direct3DCreate9' ...
6368 17:17:06.656 Creating interface m_IDirect3D9Ex::m_IDirect3D9Ex (02AFE358)
6368 17:17:06.656 m_IDirect3D9Ex::LogAdapterNames Adapter: 0 \\.\DISPLAY1 NVIDIA GeForce RTX 3080
6368 17:17:06.656 m_IDirect3D9Ex::LogAdapterNames Adapter: 1 \\.\DISPLAY2 NVIDIA GeForce RTX 3080
6368 17:17:06.656 dd_DirectDrawCreate
6368 17:17:06.656 dd_DirectDrawCreateEx
6368 17:17:06.656 Direct3D9SetSwapEffectUpgradeShim Calling 'Direct3D9SetSwapEffectUpgradeShim' ... 0
6368 17:17:06.656 Redirecting 'DirectDrawCreate' IID_IDirectDraw to --> 'Direct3DCreate9'
6368 17:17:06.656 Creating interface m_IDirectDrawX::m_IDirectDrawX (00BC14F8) converting interface from v1 to v9
6368 17:17:06.656 Redirecting 'Direct3DCreate9' ...
6368 17:17:06.659 Creating interface m_IDirect3D9Ex::m_IDirect3D9Ex (02AFE4D8)
6368 17:17:06.677 m_IDirectDrawX::~m_IDirectDrawX (00BC14F8) deleting interface!
6368 17:17:06.677 m_IDirect3D9Ex::~m_IDirect3D9Ex (02AFE4D8) deleting interface!
6368 17:17:06.677 m_IDirectDrawSurfaceX::CleanupSharedEmulatedMemory Deleting 0 emulated surfaces!
6368 17:17:06.677 m_IDirect3D9Ex::~m_IDirect3D9Ex (02AFE358) deleting interface!
6368 17:17:08.037 Direct3D9SetSwapEffectUpgradeShim Calling 'Direct3D9SetSwapEffectUpgradeShim' ... 0
6368 17:17:08.037 Redirecting 'DirectDrawCreate' IID_IDirectDraw to --> 'Direct3DCreate9'
6368 17:17:08.037 Creating interface m_IDirectDrawX::m_IDirectDrawX (00BC14F8) converting interface from v1 to v9
6368 17:17:08.037 Redirecting 'Direct3DCreate9' ...
6368 17:17:08.071 Creating interface m_IDirect3D9Ex::m_IDirect3D9Ex (02A8D708)
6368 17:17:08.084 WndProc::AddWndProc Creating WndProc instance! WND(00050B9C,WarGames,{0,0,639,479})
6368 17:17:08.094 m_IDirectDrawX::CreateD3D9Device Direct3D9 device! 640x480 refresh: 0 format: D3DFMT_X8R8G8B8 wnd: WND(00050B9C,WarGames,{0,0,639,479}) params: {640,480,D3DFMT_X8R8G8B8,1,0,0,1,WND(00050B9C,WarGames,{0,0,639,479}),0,0,D3DFMT_UNKNOWN,0x0,0,0x80000000} flags: 0x46
6368 17:17:08.476 Creating interface m_IDirect3DDevice9Ex::InitDirect3DDevice (06F7A278)
6368 17:17:08.510 Creating interface m_IDirectDrawClipper::m_IDirectDrawClipper (06FE3840)
6368 17:17:08.510 m_IDirectDrawX::CreateSurface2 Primary surface 0x0 dwFlags: 0x21 ddsCaps: 0x238, 0x0, 513
6368 17:17:08.510 Creating interface m_IDirectDrawSurfaceX::m_IDirectDrawSurfaceX (06F8C580) converting interface from v1 to v9
6368 17:17:08.510 Creating interface m_IDirectDrawSurfaceX::m_IDirectDrawSurfaceX (06F8C7F8) converting interface from v1 to v9
6368 17:17:08.510 Creating interface m_IDirectDrawPalette::m_IDirectDrawPalette (06F8CA70)
6368 17:17:08.511 Creating interface m_IDirect3DTexture9::m_IDirect3DTexture9 (06FD49B8)
6368 17:17:08.518 Creating interface m_IDirect3DSurface9::m_IDirect3DSurface9 (06FC65A8)
6368 17:17:08.532 Creating interface m_IDirect3DTexture9::m_IDirect3DTexture9 (06FD4D00)
6368 17:17:08.533 Creating interface m_IDirect3DSurface9::m_IDirect3DSurface9 (06FC6DE8)
6368 17:17:08.534 Creating interface m_IDirect3DVertexBuffer9::m_IDirect3DVertexBuffer9 (06FD4EC8)
6368 17:17:08.581 UnhandledExceptionFilter: exception code=0xc0000005 flags=0 addr=0056AC0B
6368 17:17:08.594 UnhandledExceptionFilter: NOP opcode=  len=2
6368 17:17:08.594 UnhandledExceptionFilter: exception code=0xc0000005 flags=0 addr=0056AC14
6368 17:17:08.594 UnhandledExceptionFilter: NOP opcode=  len=2

**Comment by mirh** [comment]
Does it change anything with [the patch](https://www.patches-scrolls.de/patch/4724/7)? Or does it start with dgvoodoo?

**Comment by elishacloud** [comment]
Here is an update.  I disabled some things that might cause issues on Windows 11:  [dxwrapper.zip](https://github.com/user-attachments/files/16302120/dxwrapper.zip)

If that does not work here are a few other things to try:

1. Disable any security product you are using.  It could be triggering a false positive on dxwrapper.
2. Try using only a single monitor or set the monitors up to duplicate the image on both screens.  Some games have issues with dual monitors.
3. Try disabling high DPI settings.  These can sometime interfere with some games.

**Comment by Destova** [comment]
Sadly nothing worked from this. I'm really confused why it ran perfectly smoothly that one day and then just... doesn't now. It really doesn't make sense.

**Comment by Destova** [comment]
Yeah I actually thought of that, and un-installed the update that happened a few days ago. Sadly no go.

**Comment by elishacloud** [comment]
If it is not an MS patch then it could be your antivirus update.  Did you try disabling you antivirus before running the game?

The only other thing I can think if is that it could be a corrupted file in the game.  Did you try reinstalling?

If those don't work I will have to spend some time when I have a chance and install Window 11 on a VM and see if it can be reproduced there.

**Comment by Destova** [comment]
Disabling AV did nothing sadly, still same unhandled exception in the logs and still same black screen back to desktop.

I will try to rip another install from my CD when I get home, but I pulled the same copy from my USB stick I used to move it over to my desktop PC (it doesn't have a disk drive) and tried it there, with no luck. Weirdly enough the installer from the disk refuses to run in windows 11, so I had to install it on my windows 10 system and then steal all the files from it that way via USB.

**Comment by mirh** [comment]
Take note that sometimes AV don't really shut down 100%. 
Pretty much had something identical to your problem happening with spotify, and only after uninstalling bitdefender it got solved.

**Comment by Destova** [comment]
> Take note that sometimes AV don't really shut down 100%. Pretty much had something identical to your problem happening with spotify, and only after uninstalling bitdefender it got solved.

I've only got windows defender on my system since that's really all you need nowadays unless you somehow get a major infection. I disabled everything on it and still no luck even after getting a new install of the game. I'm at a complete loss at this point.

**Comment by mirh** [comment]
Can you try to log the game with procmon, and see what it tries to look in the instants leading to the error message?

**Comment by Destova** [comment]
> Can you try to log the game with procmon, and see what it tries to look in the instants leading to the error message?

I'm sorry I'm only now seeing this, what is procmon? It's also not giving an error message on the desktop, only in the logs.

**Comment by StefanStanic95** [comment]
@Destova in case you still want to try things to get the game running, what worked for me is removing the [HotPatch] section from the dxwrapper.ini file, this entire section in specific:
```ini
[HotPatch]
VerificationAddress        = 0xFF
VerificationBytes          = 0xFF
PatternString              = 0F 85 ? ? ? ? 83 3D ? ? ? ? 06 0F 8D
BytesToWrite               = 0x90E9
PatternString              = 74 0A B8 01 00 00 00 E9 ? 00 00 00 C6 85
BytesToWrite               = 0x9090
```
Keep in mind, however, that I was not getting any unhandled exceptions in my logs.

If all else fails, try using the older version of the patch, designed for Windows 10:
https://www.gamepressure.com/download/wargames-defcon-1-windows-10-fix-mod/z5128a9

**Comment by Destova** [comment]
> [@Destova](https://github.com/Destova) in case you still want to try things to get the game running, what worked for me is removing the [HotPatch] section from the dxwrapper.ini file, this entire section in specific:
> 
> [HotPatch]
> VerificationAddress        = 0xFF
> VerificationBytes          = 0xFF
> PatternString              = 0F 85 ? ? ? ? 83 3D ? ? ? ? 06 0F 8D
> BytesToWrite               = 0x90E9
> PatternString              = 74 0A B8 01 00 00 00 E9 ? 00 00 00 C6 85
> BytesToWrite               = 0x9090
> 
> Keep in mind, however, that I was not getting any unhandled exceptions in my logs.
> 
> If all else fails, try using the older version of the patch, designed for Windows 10: https://www.gamepressure.com/download/wargames-defcon-1-windows-10-fix-mod/z5128a9

Oh. My. God. 

It worked. It's working. It got to the main menu anyway before I physically screamed out loud. Haven't tried a game yet, but you did it I think! Holy cow. Thank you!

**Comment by Destova** [comment]
I'll say as well, I had 2 [HotPatch] lines in my dxwrapper, not just the one. I deleted both though and I can confirm it now works (had to find my iso but idc about no cd cracks, I have two different ISOs from two different versions of the game anyway, including the one that came with that virus LOL).

**Comment by StefanStanic95** [comment]
Glad to hear. It seems the developer updated the WarGames-fix.zip file to no longer enable the [HotPatch] lines by default, on the same day as you tested my suggestion.

**Comment by Destova** [comment]
I'm wondering why that of all things was causing my issues? Guess I'll never really know, but as long as it keeps working I'll be happy. This game means way too much to me to have given up like I did on getting it running.

**Comment by elishacloud** [comment]
Seems like the hotpatch is modifying the wrong part of code in your version of the game.  Can you send me your exe file so I can compare it?

**Comment by Destova** [comment]
> Seems like the hotpatch is modifying the wrong part of code in your version of the game. Can you send me your exe file so I can compare it?

Tried to upload it here but github doesn't accept this, so I put it on google drive. 

This is an install from an ISO I ripped from the not 1.01 version of the disc (my 1.01 disc cracked years ago due to an accident and I got my hands on a copy from Ebay that was the older disc, it's the one that came with the virus in the registration program (which is not present in this download I assure it, and it's a virus from 1999, it's harmless today))

https://drive.google.com/file/d/1YESjxxopUHY9P5tBIyzgsgvIyuvumLJq/view?usp=sharing

**Comment by elishacloud** [comment]
That exe file works fine for me.  Can you send me a screenshot of your wargames folder?

Also, does your video folder include the "intro.dat" file?

**Comment by Destova** [comment]
![Image](https://github.com/user-attachments/assets/6dde2423-b342-4ac9-90fa-6cdaf38a0aa2)

Yeah it does, intro.dat.

**Comment by elishacloud** [comment]
Everything looks fine.  I am not sure what is happening in your case.  Your exe file looks just like mine and it works fine for me.  But at least we know it works if you disable the [HotPatch] lines.

**Comment by elishacloud** [comment]
Can you try the latest build on the [wiki](https://github.com/elishacloud/dxwrapper/wiki/WarGames)?  I think it fixes this issue.

**Comment by elishacloud** [comment]
Closing issue.  Should be fixed in the latest wiki.  Please reopen if you still have an issue.
